### DISCUSSION TITLE
Soil mass exploded at the interface

### DISCUSSION THREAD

**Discussion by Francisliii** [Anura3D/Anura3D_OpenSource#110] [Q&A] [answered]
Hi Dear community,

I am working on simulations of cone penetration in dry sand. There are many current literatures on this topic but I got stuck on one technical issue. Here shows the mesh of the model
![fac78c17697fb96e6633efc87f1c496](https://github.com/Anura3D/Anura3D_OpenSource/assets/122603534/684220c6-aa85-4d24-b0df-5ace71ce9631)

The cone is modelled with rigid body with prescribed velocity of 0.02m/s. Mesh size below the cone tip is specifically refined. Moving mesh and contact algorithm are applied. Structured mesh in the compressive zone is implemented. The density of the surcharge is modified to create 50kpa surcharge load prior to penetration. 

With or without mass scaling, soil material points from the penetrometer-soil-surcharge zone cross the border and jump into penetrometer from the very first calculation step. Eventually, the penetrometer-soil-surcharge zone exploded the material points jump out of the mesh. 

_**(First calculation step)**_
![93a4dc7abb24f849eb9f235360b0b96](https://github.com/Anura3D/Anura3D_OpenSource/assets/122603534/629cc047-1ecf-47ed-b0fe-f6f7ecdeb05f)

_**(Final calculation step)**_
![c701d6ef9420e4e1a1eec584f02da13](https://github.com/Anura3D/Anura3D_OpenSource/assets/122603534/e35e61a0-e1ef-403d-964a-bbc34ad38da5)

Regarding the calculation, quasi-static convergence is applied with damping coefficient from 0.1 to 0.3. The bulk viscosity damping is also tried with linear coefficient of 0.06 and quadratic coefficient of 1.2. 

So just wondering if there are any other thing I can try for a possible solution? 

Thanks,
Francis

**Comment by albuquerqueraniel** [comment]
Hello Francis,

Did you solve your problem? If not, did you try refining the mesh around the penetrometer shaft? How many MP there are in the region in contact with the tip? Have you tried increase the number of MP per element in that region?

Best regards,
Raniel

**Reply by ayerrocolom** [reply, marked as answer]
Dear Francis and Raniel,

The problem might arise in some cases when the contact has "sharp" transitions or kinks. The contact node at a kink doesn't have a unique normal vector because each side has a different one, and interpenetration might occur. In that case, the user needs to change the orientation of the normal vector at that point by manually changing a specific set of flags in the CPS called $$$$APPLY_CONTACT_NORMAL_CORRECTION, $$NUMBER_OF_CORRECTED_NORMALS and $$NODE_NORMAL_DATA.

Best,
Alba

**Reply by albuquerqueraniel** [reply]
Dear Alba,

Thank you!

After your answer, I defined manually the normals of the edges. I defined vertical vectors for these nodes:

![normal correction](https://github.com/Anura3D/Anura3D_OpenSource/assets/45863863/b2256225-363a-433d-a695-f483d6c2d83a)


It solved the interpenetration issue. But now I'm facing a new problem: the MPs are moving away from the cone shaft. The soil MPs are  close to the shaft at the firsts load steps. But a gap between the cone and the soil starts to form:

![load steps](https://github.com/Anura3D/Anura3D_OpenSource/assets/45863863/4a25e358-acdf-48b9-ace7-34cafbd223ff)

In the image below there is a comparasion between my model and the model presented by Yost in the Anura3D 2023 Workshop. After 4.95 s, the gap is almost the full lenght of the penetrated shaft. But in Yost, and yours, model there isn't a gap between the shaft and the soil.

![comparasion](https://github.com/Anura3D/Anura3D_OpenSource/assets/45863863/c7f2c91e-3c31-478e-b4ef-e596f7885d16)

This gap was a problem in your model that you solved? By any chance do you know how may I solve it?

Ps.: My cone has 18" (0.4572 m) radius and its velocity is 0.5 m/s. The original geometry of my problem isn't of a cone, this geometry is a simplification. My aim is to model a tube with 1,5 " thickness, I will model my real geometry after the simulation with the cone geometry works.

Thank you in advance.
Best, 
Raniel

**Reply by ayerrocolom** [reply]
@kmyost, do you recall the orientation of the normal at the transition point you used in your models?

**Reply by kmyost** [reply]
Hello @albuquerqueraniel,

I didn't see this issue with such a large gap. I suggest trying only using the manually-defined normal at the very tip of the cone, not at the shoulder, and see if that resolves it. Orientation of the normal may need to be adjusted depending on the model/mesh. Usually, something just slightly off from vertical worked well for me.

**Reply by kmyost** [reply]
@albuquerqueraniel To clarify, I didn't have a sharp interface at the shoulder of the cone - I slightly smoothed the corner there to prevent the issue with the normals. You could try to implement that and see if it works for you!
